### PR TITLE
feat: list support cssVar

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- #45618

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | list support css variable theme |
| 🇨🇳 Chinese | 列表支持 css 变量主题 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 957e6e2</samp>

This pull request enhances the List component to support CSS variables for theming. It introduces a new `useCSSVar` hook that registers and applies the theme tokens as CSS variables. It also refactors the List style to use the `unit` and `token` methods from the `@ant-design/cssinjs` package.

### <samp>🤖 Generated by Copilot at 957e6e2</samp>

*  Add `useCSSVar` hook to register and apply CSS variables for List component based on theme tokens ([link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-015bf7609c9e2715aacae53f1bbf8779c3f64b720462e1efc033b8d2711cd9e2R16), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-015bf7609c9e2715aacae53f1bbf8779c3f64b720462e1efc033b8d2711cd9e2L146-R148), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-015bf7609c9e2715aacae53f1bbf8779c3f64b720462e1efc033b8d2711cd9e2L287-R289), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-414ef311d1d3a1bb819449722ba7262737028220ad44180298ea0280a39c53d8R1-R4))
*  Use `unit` function to convert numeric values to CSS units in List styles ([link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L1-R4), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L83-R83), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L90-R90), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L147-R147), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L254-R254), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L286-R286), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L302-R306), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L310-R314), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L364-R368), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L374-R378), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L382-R389), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L392-R396))
*  Add `prepareComponentToken` function to return default token values for List component in CSS units ([link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4R415-R430), [link](https://github.com/ant-design/ant-design/pull/45795/files?diff=unified&w=0#diff-8ed8eddb13b4537df4b99afb831a6d4be1bef5a7d3b3cc4ddac6b3d1307e87a4L422-R442))

### CODE REVIEW

File: components/list/style/index.ts [hunk 1]
```diff
@@ -65,6 +65,7 @@ export interface ComponentToken {
 interface ListToken extends FullToken<'List'> {
   listBorderedCls: string;
   minHeight: number;
+  itemActionSplitHeight: number | string;
```
Comment (MadCcc) [inline components/list/style/index.ts:68]: component token 声明在 ComponentToken 下哈

File: components/list/style/index.ts [hunk 2]
```diff
@@ -299,15 +301,15 @@ const genBaseStyle: GenerateStyle<ListToken> = (token) => {
             insetBlockStart: '50%',
             insetInlineEnd: 0,
             width: lineWidth,
-            height: Math.ceil(token.fontSize * token.lineHeight) - token.marginXXS * 2,
+            height: itemActionSplitHeight,
```
Comment (MadCcc) [inline components/list/style/index.ts:304]: fontSize * lineHeight 可以用 fontHeight 替换了，不需要提 component token 了

Comment (kiner-tang) [reply components/list/style/index.ts]: done

File: components/list/style/index.ts [hunk 3]
```diff
@@ -299,15 +299,15 @@ const genBaseStyle: GenerateStyle<ListToken> = (token) => {
             insetBlockStart: '50%',
             insetInlineEnd: 0,
             width: lineWidth,
-            height: Math.ceil(token.fontSize * token.lineHeight) - token.marginXXS * 2,
+            height: token.fontHeight,
```
Comment (MadCcc) [inline components/list/style/index.ts:302]: 后面的`- token.marginXXS * 2` 要加上哈

Comment (kiner-tang) [reply components/list/style/index.ts]: done

File: components/list/style/index.ts [hunk 4]
```diff
@@ -299,15 +299,18 @@ const genBaseStyle: GenerateStyle<ListToken> = (token) => {
             insetBlockStart: '50%',
             insetInlineEnd: 0,
             width: lineWidth,
-            height: Math.ceil(token.fontSize * token.lineHeight) - token.marginXXS * 2,
+            height: token
+              .calc(token.fontHeight)
+              .sub(token.calc(token.marginXXS).mul(2).equal())
```
Comment (MadCcc) [inline components/list/style/index.ts:304]:
```suggestion
              .sub(token.calc(token.marginXXS).mul(2))
```

**Review by MadCcc — APPROVED** [review]

**Pull request merged**